zerver/views/messages.py

```diff
@@ -424,6 +424,11 @@ def highlight_string(text, locs):
     in_tag = False
     for loc in locs:
         (offset, length) = loc
+        if not settings.USING_PGROONGA:
+            prefix = force_bytes(text)[:offset]
+            match = force_bytes(text)[offset:offset + length]
+            offset -= len(prefix) - len(force_text(prefix))
```

Comment:
All this byte arithmetic his feels slightly overcomplicated to me.  I'm wondering if we just want to make two functions here, one for each search engine, so that we can consistently work in either utf8 space (tsearch) or string space (pgroonga).  We also want to explicitly encode stuff as utf8.  Now that we're no longer straddling python2/3, I think `s.encode('utf8')` is more canonical.

~~~
>>> s = u'<p>昨日'
>>> force_bytes(s)
b'<p>\xe6\x98\xa8\xe6\x97\xa5'
>>> s.encode('utf8')
b'<p>\xe6\x98\xa8\xe6\x97\xa5'

Reply:
On this front, I think what we actually want to do is a pass through the codebase replacing `force_bytes`, `force_str`, and `force_text` with the appropriate encode/decode functions.

Reply:
I added #7240 for the general code sweep.  I think this might be a good place to do it now, since we're really doing utf8-specific stuff, as opposed to just working with a dumb library.